folly/detail/Clock.cpp

```diff
@@ -48,7 +48,125 @@ int clock_getres(clockid_t clk_id, struct timespec* ts) {
 
   return 0;
 }
-#elif _MSC_VER
+#elif defined(_MSC_VER)
+// The MSVC version has been extracted from the pthreads implemenation here:
+// https://github.com/songdongsheng/libpthread
+// Copyright(c) 2011, Dongsheng Song <songdongsheng@live.cn>
+//
+// It is under the Apache License Version 2.0, just as the rest of the file is.
```

Comment:
I don't want to go into legalese as I'm not a lawyer, but as I understand it, the only Apache licensed code we can use is code which the actual copyright holder gives us explicit rights to, and I believe the code already in this file falls under that umbrella.

This new code comes from another party.  If you can get him to sign our CLA and contribute it under those terms, we can take it.  Otherwise.... sigh... :(